container/lxd/manager.go

```diff
@@ -227,27 +224,42 @@ func (manager *containerManager) createInstance(
 	cfg := map[string]string{
 		UserDataKey:      string(userData),
 		NetworkConfigKey: cloudinit.CloudInitNetworkConfigDisabled,
-		// An extra piece of info to let people figure out where this
-		// thing came from.
-		JujuModelKey: manager.modelUUID,
-		// Make sure these come back up on host reboot.
-		AutoStartKey: "true",
+		AutoStartKey:     "true",
+		// Extra info to indicate the origin of this container.
+		JujuModelKey: m.modelUUID,
 	}
 
-	nics, err := networkDevices(networkConfig)
+	nics, unknown, err := networkDevicesFromConfig(networkConfig)
 	if err != nil {
 		return "", errors.Trace(err)
 	}
 
 	var profiles []string
 	if len(nics) == 0 {
-		logger.Infof("instance %q configured with %q profile", name, lxdDefaultProfileName)
+		logger.Infof("configuring container %q with %q profile", name, lxdDefaultProfileName)
 		profiles = []string{lxdDefaultProfileName}
 	} else {
-		logger.Infof("instance %q configured with %v network devices", name, nics)
+		logger.Infof("configuring container %q with network devices: %v", name, nics)
+
+		// If the default LXD bridge was supplied in network config,
+		// but without a CIDR, attempt to ensure it is configured for IPv4.
+		// If there are others with incomplete info, log a warning.
+		if len(unknown) > 0 {
+			if len(unknown) == 1 && unknown[0] == network.DefaultLXDBridge && m.server.networkAPISupport {
```

Comment:
I'm a little uncomfortable about "unknown[0] == DefaultLXDBridge' as that forces us to only really work well if there is an lxdbr0. But for now we can deal. It just feels like we should be trying to get rid of special cases as much as we can.

Reply:
Yes; all this is going to need some careful thought around the networking changes for clustering.

In this case, my thought was that this is the situation that most likely represents the LP bug scenario and that we can be reasonably safe with the modification.

If the default profile has some custom networking configuration, it is unlikely to be seen as helpful for Juju to modify it automatically.